File: python/01_broken_access_control.py
"""A01 - Broken Access Control. VULN: IDOR via unchecked user_id."""
# VULN: A01 - any caller can read any user's profile by changing user_id
def get_profile(db, user_id, requester_id):
    # missing: if user_id != requester_id and not is_admin(requester_id): deny
    return db.execute(f"SELECT * FROM users WHERE id = {user_id}")

File: javascript/01_broken_access_control.js
// A01 - Broken Access Control. VULN: missing ownership check on delete.
// VULN: A01 - IDOR delete
function deleteDocument(db, docId, userId) {
  return db.run(`DELETE FROM docs WHERE id = ${docId}`);
  // missing: AND owner_id = userId
}

module.exports = { deleteDocument };
